android/java/org/chromium/chrome/browser/brave_news/CardBuilderFeedCard.java

```diff
@@ -0,0 +1,1412 @@
+            @Override
+            public boolean onLoadFailed(@Nullable GlideException e, Object model, Target target,
+                    boolean isFirstResource) {
+                return false;
+            }
+            @Override
+            public boolean onResourceReady(Drawable resource, Object model, Target target,
+                    DataSource dataSource, boolean isFirstResource) {
+                if (resource instanceof BitmapDrawable) {
+                    Bitmap bitmap = ((BitmapDrawable) resource).getBitmap();
+                    // Log.d(TAG,String.format(Locale.getDefault(), "Ready %s bitmap %,d bytes,
+                    // size: %d x %d",
+                    //         name,
+                    //         bitmap.getByteCount(),
+                    //         bitmap.getWidth(),
+                    //         bitmap.getHeight()));
+                }
+                return false;
+            }
+        };
+    }
+
+    private LayerDrawable makeRound(int type, int background, int radius) {
+        float[] outerRadii =
+                new float[] {radius, radius, radius, radius, radius, radius, radius, radius};
+        float[] innerRadii =
+                new float[] {radius, radius, radius, radius, radius, radius, radius, radius};
+
+        // Set the shape background card_background
+        ShapeDrawable backgroundShape = null;
+        new ShapeDrawable(new RoundRectShape(outerRadii, null, innerRadii));
+
+        if (type == BUTTON_LAYOUT) {
+            backgroundShape = new ShapeDrawable(new RoundRectShape(outerRadii, null, innerRadii));
+            backgroundShape.getPaint().setColor(
+                    mActivity.getResources().getColor(background)); // background color
+            backgroundShape.getPaint().setStyle(Paint.Style.FILL); // Define background
+            backgroundShape.getPaint().setAntiAlias(true);
+            backgroundShape.setPadding(10, 10, 10, 10);
+
+        } else if (type == CARD_LAYOUT) {
+            backgroundShape = new ShapeDrawable(new RoundRectShape(outerRadii, null, innerRadii));
+            backgroundShape.getPaint().setColor(
+                    mActivity.getResources().getColor(background)); // background color
+            backgroundShape.getPaint().setStyle(Paint.Style.FILL); // Define background
+        } else if (type == ROUNDED_TOP_LAYOUT) {
+            backgroundShape = new ShapeDrawable(new RoundRectShape(outerRadii, null, null));
+            backgroundShape.getPaint().setColor(
+                    mActivity.getResources().getColor(background)); // background color
+            backgroundShape.getPaint().setStyle(Paint.Style.FILL); // Define background
+        }
+
+        // Initialize an array of drawables
+        Drawable[] drawables = new Drawable[] {backgroundShape};
+        // Initialize a layer drawable object
+        LayerDrawable layerDrawable = new LayerDrawable(drawables);
+        return layerDrawable;
+    }
+}
```

Comment:
potential nit for follow up: This file is pretty long - perhaps break up, e.g. card-specific decisions vs more general drawing or data gathering?

Reply:
yeah for sure